cli/src/infragpt/api_client.py

```
@@ -0,0 +1,182 @@
+                        message = (
+                            error_data.get("error")
+                            or error_data.get("message")
+                            or response.text
+                        )
+                    except Exception:
+                        message = response.text or f"HTTP {response.status_code}"
+                    raise InfraGPTAPIError(response.status_code, message)
+
+                return response.json()
+        except httpx.TimeoutException:
+            raise InfraGPTAPIError(0, "Request timed out")
```

Ruff (0.14.10)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)